lib/amplitude_api/event.rb

```diff
@@ -81,11 +86,17 @@ def to_hash
       serialized_event[:user_id] = user_id
       serialized_event[:event_properties] = event_properties
       serialized_event[:user_properties] = user_properties
+      serialized_event = add_optional_properties(serialized_event)
```

Comment:
rubocop complained when I added one more `serialized_event[:distinct_id] if distinct_id` to the method. so it seemed to make the most sense to pull out all the optional properties into their own method, vs telling rubocop to ignore that method.